library/gui/key_bindings/src/main/java/org/quiltmc/qsl/key/bindings/impl/KeyBindingRegistryImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.key.bindings.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+import net.minecraft.client.option.GameOptions;
+import net.minecraft.client.option.KeyBinding;
+
+import org.quiltmc.qsl.key.bindings.mixin.client.KeyBindingAccessor;
+
+@Environment(EnvType.CLIENT)
+public class KeyBindingRegistryImpl {
+	public static final Logger LOGGER = LogManager.getLogger();
+
+	private static Map<KeyBinding, Boolean> quiltKeys = new HashMap<>();
+	private static List<KeyBindingManager> keyBindingManagers = new ArrayList<>(1);
+	private static KeyBinding[] enabledQuiltKeysArray = new KeyBinding[] {};
+	private static List<KeyBinding> disabledQuiltKeys = new ArrayList<>(0);
+
+	public static KeyBinding registerKeyBinding(KeyBinding key, boolean enabled) {
+		if (key == null) {
+			LOGGER.error("Attempted to register a null key bind!");
```

Comment:
Why is this just logged and not an immediate exception?

Reply:
Like, throw an NPE? I already return it immediately after that message

Reply:
You just return `null`, not an NPE.
This would mean that an NPE is thrown the first time the mod tries to use the keybind, which is... not optimal.

Reply:
I already return `null` one line after the log message though, I don't let the rest of the code run and make bad things happen

Reply:
Oh boy, that was a confusing discussion, but the idea (use exceptions instead of returning null) has been implemented